cumulus/test/runtime/src/test_pallet.rs

```diff
@@ -121,3 +126,27 @@ pub mod pallet {
 		}
 	}
 }
+
+impl<T: Config> cumulus_pallet_parachain_system::OnSystemEvent for Pallet<T> {
+	fn on_validation_data(_data: &cumulus_primitives_core::PersistedValidationData) {
+		// Nothing to do here for tests
+	}
+
+	fn on_validation_code_applied() {
+		// Nothing to do here for tests
+	}
+
+	fn on_relay_state_proof(
+		relay_state_proof: &cumulus_pallet_parachain_system::relay_state_snapshot::RelayChainStateProof,
+	) -> frame_support::weights::Weight {
+		use crate::test_pallet::RELAY_EPOCH_INDEX_KEY;
+
+		// Expect the requested key to be part of the proof.
+		relay_state_proof
+			.read_optional_entry::<u64>(RELAY_EPOCH_INDEX_KEY)
```

Comment:
My point was to use a key that isn't by default part of the proof, this is already part of the proof. Let's take the balance key for ALICE for example.